frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,753 @@
+}
+
+function reducer(state: FormState, action: FormAction) {
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: null,
+        },
+        submitError: null,
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      const hasNoErrors = Object.values(state.validationErrors).every(
+        (prop) => prop == null || prop === ""
+      );
+
+      return {
+        ...state,
+        status: hasNoErrors ? FormStatus.Submitting : FormStatus.Idle,
+      };
+    }
+
+    case FormActionType.Success:
+      return {
+        // Display thanks page
+        ...state,
+        status: FormStatus.Submitted,
+      };
+
+    case FormActionType.Failed:
+      return {
+        // Resta in idle, setta errore e resetta i campi?
+        ...state,
+
+        submitError: "Error while sending your message",
+        status: FormStatus.Idle,
+      };
+
+    case FormActionType.ValidationFailed:
+      return {
+        ...state,
+        validationErrors: action.validationErrors,
+      };
+
+    default: {
+      assertNever(action);
+    }
+  }
+}
+
+export function ContactsForm() {
+  const [state, dispatch] = React.useReducer(reducer, blankForm);
+
+  function validateErrors(values: FormState["values"]) {
+    const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+
+    const validationErrors: FormState["validationErrors"] = {
+      name: null,
+      email: null,
+      message: null,
+    };
+
+    if (values.name.trim().length === 0) {
+      validationErrors.name = "Please insert your name";
+    } else {
+      validationErrors.name = null;
+    }
+
+    if (!regex.test(state.values.email)) {
+      validationErrors.email = "Please insert a valid email";
+    } else {
+      validationErrors.email = null;
+    }
+    if (state.values.message.trim().length == 0) {
+      validationErrors.message = "Please insert a message";
+    } else {
+      validationErrors.message = null;
+    }
+    return validationErrors;
+  }
+
+  React.useEffect(() => {
+    if (state.status === FormStatus.Submitting) {
+      async function sendMessage() {
+        const input: CreateFormMessageInput = {
+          data: {
+            name: state.values.name,
+            email: state.values.email,
+            message: state.values.message,
+          },
+        };
+        const response = await fetchGraphQL<
+          InsertFormMessageMutation,
+          InsertFormMessageMutationVariables
+        >(InsertFormMessage, { input });
+
+        if (response.createFormMessage == null) {
+          dispatch({ type: FormActionType.Failed });
+        } else {
+          dispatch({ type: FormActionType.Success });
+        }
+      }
+      sendMessage();
+    }
+  }, [state.status, state.values]);
```

Comment:
The issue here, is that if `state.values` changes while `state.status === FormStatus.Submitting` then the request will be triggered multiple times. As we've already discussed, the fetch request should be made from the `onFormSubmit` event handler.. The pseudocode is something like this

```
onFormSubmit()
BEGIN
check for validation errors
IF there are validation errors
  update state with validation errors
ELSE
  update state to indicate that you are submitting the form
  TRY
    make the fetch request
    update the state to indicate that the request has succeded
  CATCH
    update the state to indicate that the requst has failed
END
```